zk/stages/stage_sequence_execute.go

```diff
@@ -141,15 +141,17 @@ func sequencingBatchStep(
 	streamWriter := newSequencerBatchStreamWriter(batchContext, batchState)
 
 	// injected batch
-	if executionAt == 0 {
-		if err := processInjectedInitialBatch(batchContext, batchState); err != nil {
+	// TODO: check if we don't need injected batch in normalcy
+	if executionAt == 0 && !cfg.chainConfig.IsNormalcy(executionAt) {
```

Comment:
we have our answer for this now.  Normalcy mode for pessimistic proofs does have an injected batch but it is read from disk rather than being read from the L1.  So this logic should stay in place and we need to look into how the config is read from disk, I assume it's something in the flags.